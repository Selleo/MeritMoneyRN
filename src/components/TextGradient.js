import React, { Component } from 'react'
import { LinearTextGradient } from 'react-native-text-gradient'

import { secondaryLight, primary } from 'src/styles/colors'

export default class TextGradient extends Component {
  render() {
    const { style, children } = this.props

    return (
      <LinearTextGradient
        colors={[secondaryLight, primary]}
        end={{ x: 1, y: 0 }}
        locations={[0, 1]}
        start={{ x: 0, y: 0 }}
        style={style}
      >
        {children}
      </LinearTextGradient>
    )
  }
}
