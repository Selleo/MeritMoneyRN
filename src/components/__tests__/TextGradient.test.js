import 'react-native'
import React from 'react'
import renderer from 'react-test-renderer'

import TextGradient from '../TextGradient'

describe('<TextGradient />', () => {
  const defaultProps = {
    children: 'test text',
    style: {},
  }

  const wrapper = renderer.create(<TextGradient {...defaultProps} />)

  test('render', () => {
    expect(wrapper).toMatchSnapshot()
  })
})
