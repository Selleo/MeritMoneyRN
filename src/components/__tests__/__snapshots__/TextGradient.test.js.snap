// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextGradient /> render 1`] = `
<RNLinearTextGradient
  accessible={true}
  allowFontScaling={true}
  colors={
    Array [
      4282147443,
      4281317757,
    ]
  }
  ellipsizeMode="tail"
  gradientEnd={
    Object {
      "x": 1,
      "y": 0,
    }
  }
  gradientStart={
    Object {
      "x": 0,
      "y": 0,
    }
  }
  locations={
    Array [
      0,
      1,
    ]
  }
  style={
    Array [
      Object {
        "color": "gray",
      },
      Object {},
    ]
  }
>
  test text
</RNLinearTextGradient>
`;
